plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2053,6 +2078,43 @@ static auto maybe_make_debug_time_logger() -> std::optional<decltype(make_debug_
    }
 }
 
+void block_only_sync::schedule() {
+   if (!_pending) {
+      // wait one second to see if we can actually get the block from net plugin before we try to resync from block vault
+      _start_sync_timer.expires_from_now(boost::posix_time::seconds(1));
+      _pending = true;
+      _start_sync_timer.async_wait(
+          app().get_priority_queue().wrap(priority::high, [this](const boost::system::error_code& ec) {
+             if (!ec) {
+                auto id = _impl->chain_plug->chain().last_irreversible_block_id();
+                fc_dlog(_log, "Attempt to resync from block vault");
+                try {
+                  _impl->blockvault->sync(&id, *this);
+                } catch( fc::exception& er ) {
+                   wlog("Attempting to resync from blockvault encountered ${details}; the node must restart to "
+                        "continue!",
+                        ("details", er.to_detail_string()));
+                   app().quit();
+                }
+             }
+             _pending = false;
+          }));
+   }
+}
+
+void block_only_sync::on_snapshot(const char*) {
+   EOS_THROW(producer_exception, "a snapshot");
+}
+
+void block_only_sync::on_block(eosio::chain::signed_block_ptr block) {
+   try {
+      _impl->on_sync_block(block,  block->block_num() != _impl->chain_plug->chain().head_block_num() + 1);
```

Comment:
__(Not necessarily a change request, but more of a strong comment)__
What is the reason for having the logic for `check_connectivity` live outside of the function?

This seems like an implementation detail of `on_sync_block` that is spilling over.  Plus you already have all of the pieces to evaluate this as variables on `on_sync_block` (`chain` and `blk_num`).

Reply:
I didn't write that function `on_sync_block` . However, if if bart agrees that the usage of the function in `blockvault_sync_strategy.hpp:91` where it checks `_received_snapshot` is unnecessary, I can change it.

Reply:
We can internalize the check to `on_sync_block` but it will do more than the boolean passed in from the outside caller that has more information.  It looks trivial because this logic is not correct. 

We should bring in @ndcgundlach  and discuss.